From 3db42b58f2885b95d11f4b9c391440d5f4e9fcd2 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Mon, 3 Feb 2025 20:02:30 +0100
Subject: [PATCH] sunxi: H616: dram: Lower max row count

18 rows would mean pretty big RAM size, more than it's supported by
hardware. Lower it to 17 to avoid misdetection of DRAM size.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 arch/arm/mach-sunxi/dram_sun50i_h616.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-sunxi/dram_sun50i_h616.c b/arch/arm/mach-sunxi/dram_sun50i_h616.c
index cbf4668b406..76c76d2f757 100644
--- a/arch/arm/mach-sunxi/dram_sun50i_h616.c
+++ b/arch/arm/mach-sunxi/dram_sun50i_h616.c
@@ -1380,12 +1380,12 @@ static void mctl_auto_detect_dram_size(const struct dram_para *para,
 	debug("detected %u columns\n", config->cols);
 
 	/* reconfigure to make sure that all active rows are accessible */
-	config->rows = 18;
+	config->rows = 17;
 	mctl_core_init(para, config);
 
 	/* detect row address bits */
 	shift = config->bus_full_width + 4 + config->cols;
-	for (config->rows = 13; config->rows < 18; config->rows++) {
+	for (config->rows = 13; config->rows < 16; config->rows++) {
 		if (mctl_mem_matches(1ULL << (config->rows + shift)))
 			break;
 	}
